rllib/core/models/torch/primitives.py

```diff
@@ -56,3 +62,106 @@ def __init__(
 
     def forward(self, x):
         return self.mlp(x)
+
+
+class TorchCNN(nn.Module):
+    """A model containing a CNN and a final linear layer."""
+
+    def __init__(
+        self,
+        input_dims: Union[List, Tuple] = None,
+        filter_specifiers: List[List[Union[int, List]]] = None,
+        filter_layer_activation: str = "relu",
+        output_activation: str = "linear",
+        output_dim: int = None,
+    ):
+        """Initialize a TorchCNN object.
+
+        Args:
+            input_dims: The input dimensions of the network.
+            filter_specifiers: A list of lists, where each element of an inner list
+                contains elements of the form
+                `[number of filters, [kernel width, kernel height], stride]` to
+                specify a convolutional layer stacked in order of the outer list.
+            filter_layer_activation: The activation function to use after each layer (
+                except for the output).
+            output_activation: The activation function to use for the output layer.
+            output_dim: The output dimensions are [BATCH, output_dim]. We append a final
+                convolutional layer depth-only filters that is flattened and a final
+                linear layer to achieve this.
+        """
+        super().__init__()
+
+        assert len(input_dims) == 3
+        assert isinstance(output_dim, int)
+        assert filter_specifiers is not None, "Must provide filter specifiers."
+
+        filter_layer_activation = get_activation_fn(
+            filter_layer_activation, framework="torch"
+        )
+
+        output_activation = get_activation_fn(output_activation, framework="torch")
+
+        layers = []
+        # Create some CNN layers
+        core_layers = []
+
+        # Add user-specified hidden convolutional layers first
+        width, height, in_depth = input_dims
+        in_size = [width, height]
+        for out_depth, kernel, stride in filter_specifiers:
+            padding, out_size = same_padding(in_size, kernel, stride)
+            # TODO(Artur): Inline SlimConv2d
+            core_layers.append(
+                SlimConv2d(
```

Comment:
Why do we want to use these layers, instead of the built-int Conv2D? What are we adding on top of it? Is it taking away the complexity of Conv2D? if not I would argue for removing one extra layer and directly link to Conv2D here.

Reply:
Just like further up (https://github.com/ray-project/ray/pull/32515#discussion_r1113616936), I did it simply to share more code with the old Models until these are gone. Just to make it easier to not deviate from their behaviour!